extensions/github1s/src/github1sfs.ts

```diff
@@ -565,4 +577,58 @@ export class GitHub1sFS
 			return null;
 		});
 	}
+
+	provideTextSearchResults(
```

Comment:
Should we split the `class GitHub1sFS` into multiple files by the functionality, like:

```js

// file: GitHub1sFileSystemProvider.ts
export class GitHub1sFileSystemProvider implements FileSystemProvider, Disposable {
  ...
}

// file: GitHub1sFileSearchProvider.ts
export class GitHub1sFileSearchProvider implements FileSearchProvider, Disposable {
  ...
}

// file: GitHub1sFileDecorationProvider.ts
export class GitHub1sFileDecorationProvider implements FileDecorationProvider, Disposable {
  ...
}

// file: GitHub1sTextSearchProvider.ts
export class GitHub1sTextSearchProvider implements TextSearchProvider, Disposable {
  ...
}
```

So that the functions would not be in a single file and it's easier for long-term maintenance and other contributors.

Reply:
Yes, this file is seems to large, I will split it next